Sprint 14

anton-kashnikoff opened the pull request.

Commit:
Changed the deployment target to iOS 13.4
Deleted unsupported devices platforms and orientations

Commit:
Deleted unneeded comments

Commit:
Deleted the main storyboard

Commit:
Added the window and collection view controller as a root view controller

Commit:
Added base code for working with collection view

Commit:
Added "add tracker" button and date picker to navigation bar

Commit:
Added search text field

Commit:
Realized basic layout and interface of the main screen

Commit:
Added tab bar controller

Commit:
Changed label text font size

Commit:
Renamed class of collection view controller

Commit:
Added launch screen

Commit:
Added second tab bar item

Commit:
Added extension to UIColor

Commit:
Renamed the file for extension to UIColor

Commit:
Added structs for storing data models

Commit:
Added TrackerTypeViewController

Commit:
Added NewHabitViewController with text field

Commit:
Added table view

Commit:
Deleted unused files

Commit:
Added emoji collection view

Commit:
Added emoji label for emoji collection view

Commit:
Added label for color collection view

Commit:
Added color collection view

Commit:
Added scroll view

Commit:
Added create and cancel buttons

Commit:
Added restriction for text field (38 symbols maximum)

Commit:
Added clear button to the text field while editing

Commit:
Added Category view controller

Commit:
Added New Category view controller

Commit:
Reorganizisation of files and folders

Commit:
Added Schedule view controller

Commit:
Added brief days of the week

Commit:
Fixed bugs with blinding button
Fixing constraints of scroll view
Deleted last separators of table views

Commit:
Fixed color of add tracker button in dark mode

Commit:
Fixed text button color in dark mode when choose type of tracker

Commit:
Added selection of items in collections views

Commit:
Fixed bug when adding view category didn't show new category in table view
Added saving information about habit tracker

Commit:
Added activate create button method to change button color when ready to create tracker

Commit:
Added border to tab bar

Commit:
Added action for cancel button

Commit:
Changed search text field color in dark mode

Commit:
Refactoring (change delegate of collection views)

Commit:
Change delegate of trackers collection view

Commit:
Changed logic to save data about trackers
Added tracker collection view cell

Commit:
Now when create tracker button tapped dismissed all modal view controllers

Commit:
Added displaying of tracker collection view
Changed navigation animation between view controllers (present -> push)
Changed plus button of the collection view cell

Commit:
Added data helper class to add trackers into categories

Commit:
Added Martin the tracker as completed

Commit:
Fixed bug with incorrect days count label

Commit:
Prepared to display trackers for current date

Commit:
Added displaying trackers for current date

Commit:
Fixed bug when color of completed button didn't changed

Commit:
Added label "Каждый день" when selected all days for tracker

Commit:
Fixed searching trackers

Commit:
Changed image when nothing found

Commit:
Added dismissing keyboard when tapping outside text field

Commit:
Added marking trackers as completed (fixed bugs)

Commit:
Added more mock data

Commit:
Fixed bug when trackers for current date didn't display
Fixed bug when search field is empty but trackers for this date didn't display

Dmitrii-Diadiushkin commented on Tracker/CustomClasses/TrackersCollectionView.swift:
```diff
@@ -0,0 +1,129 @@
+    }
+}
+
+extension TrackersCollectionView: UICollectionViewDelegate {
```
Если методы делегата не используется, то лучше его и не объявлять

Dmitrii-Diadiushkin commented on Tracker/Models/Schedule.swift:
```diff
@@ -0,0 +1,42 @@
+        }
+    }
+    
+    enum BriefDayOfWeek: String, CaseIterable {
```
а еще можно было не объявлять новый enum, а написать метод в DayOfWeek, который бы возвращал сокращенное название

Dmitrii-Diadiushkin commented on Tracker/Extensions/Date+Extensions.swift:
```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+extension Date {
+    public var withRemovedTime: Date {
```
название переменной вводит в заблуждение - метод .startOfDay у структуры календарь не убирает время, а возвращает дату со временем 0:0:00

Dmitrii-Diadiushkin commented on Tracker/ViewControllers/TrackersViewController.swift:
```diff
@@ -0,0 +1,245 @@
+    var categories = [TrackerCategory]()
+    var currentDate = Date()
+    var dataHelper: DataHelper?
+    var categoriesToShow = [TrackerCategory]()
```
непонятно зачем вводить эту и следующую переменные. достаточно ввести одну и наполнять ее при смене дат или вводе в строку поиска. И из неё обновлять коллекцию

Dmitrii-Diadiushkin commented on Tracker/ViewControllers/TrackersViewController.swift:
```diff
@@ -0,0 +1,245 @@
+        ])
+    }
+    
+    private func showTrackersForDate(_ date: Date) {
```
при вызове этого метода ты передаешь только currentDate - свойство самого класса. можно просто использовать это свойство внутри метода

Commit:
Deleted unused extension
Renamed name of method for zero time
Removed useful enum and created function to get brief day of the week

Commit:
Fixed distance between header and collection view cell

Commit:
Fixed weight of title font in navigation bar

Commit:
Added cancel button to search text field

Commit:
Imporoved algorithm of searching and date filtering
Fixed bugs

Commit:
Fixed absence of logo on the launch screen
Added app icon

Commit:
Added custom table view cell for displaying categories

Commit:
Added hiding keyboard when tap return

Commit:
Fixed retain cycles
